spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -167,6 +167,26 @@ public long executePartitionedDmlStatement(Statement statement) {
 		return read(entityClass, key, null);
 	}
 
+	@Override
+	public <T> boolean existsById(Class<T> entityClass, Key key) {
+		Assert.notNull(key, "A non-null key is required.");
+
+		SpannerPersistentEntity<?> persistentEntity = this.mappingContext.getPersistentEntity(entityClass);
+		String columnName = getPrimaryKeyColumnName(persistentEntity);
+		KeySet keys = KeySet.singleKey(key);
+
+		ResultSet resultSet = executeRead(persistentEntity.tableName(), keys, Collections.singleton(columnName), null);
+		maybeEmitEvent(new AfterReadEvent(Collections.emptyList(), keys, null));
+		return resultSet.next();
+	}
+
+	private String getPrimaryKeyColumnName(SpannerPersistentEntity<?> persistentEntity) {
```

Comment:
Do we use this anywhere else? Seems like it would be a common thing.

Reply:
Never. In many cases, the Key is a set of columns but for this particular case, we should know the only first primary to try to fetch it. On success, it will mean that the entity exists.

It depends on a Database but the query "IF EXISTS.." runs faster because it will not fetch a real data from real disk but only some index that probably already in-memory cache. But I'm not sure that such an assumption is really actual for Spanner :) - we could try to fetch a real column.